packages/browser-wallet/src/popup/shell/Root.tsx

```diff
@@ -1,14 +1,23 @@
-import React from 'react';
+import { Provider } from 'jotai';
+import React, { Suspense } from 'react';
+import { useTranslation } from 'react-i18next';
 import { MemoryRouter } from 'react-router-dom';
 
 import './i18n';
 
 import Routes from './Routes';
 
 export default function Root() {
+    const { t } = useTranslation();
+
     return (
-        <MemoryRouter>
-            <Routes />
-        </MemoryRouter>
+        <Provider>
+            <MemoryRouter>
+                {/* The Suspense is needed here due to async atoms loading stuff from storage */}
+                <Suspense fallback={t('root.loading')}>
```

Comment:
This suspense looks like dark magic to me :mage:

Reply:
it is 🧙‍♂️